### ISSUE TITLE
Smoothieware Connection via telnet port 23 disconnection

### ISSUE THREAD

**Issue by augustenlee** [arkypita/LaserGRBL#1599]
Hi sir,

Good day, and I am so glad to have this change to report you this:
I got the latest Smoothieware firmware (using CNC mode) and try to make connection to board via telnet(port 23).
Once I button down the "connect" , but it disconnected immediately.
after I read the log file , there was something strange while system parsing the MACHINE POSITION.
attached pictures for you that you will understand what my question is ...

the response from smoothie ware board is: 
![截圖 2021-10-05 下午10 28 02](https://user-images.githubusercontent.com/15098382/136048780-753dbba3-bad8-4b78-904d-104036c51c48.png)

and the SmoothieCore.cs is:

![截圖 2021-10-05 下午10 28 52](https://user-images.githubusercontent.com/15098382/136049088-e769eecf-5fc7-4b5f-8fc6-899e2c61d59e.png)

I want to ask that if I can make a little modification of the code if you agree it ,, like it:
![截圖 2021-10-05 下午10 42 12](https://user-images.githubusercontent.com/15098382/136049452-381e731e-955b-4db5-86c8-6b082e6ab8e6.png)

thank you very much!

augustenlee@gmail.com

**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Hardware and software configuration (please complete the following information):**
 - Windows version
 - Engraver brand and model
 - Grbl version (i.e. 0.9j, 1.1f)
 - LaserGRBL version

**Additional context**

**Comment by arkypita** [comment]
The code of the ParseF function is correct as it acts only on the F: 400.0,102.0 part of the string.
There is no reason to change it.

> after I read the log file , there was something strange while system parsing the MACHINE POSITION.

Post the error message you find in the log file.

**Comment by augustenlee** [comment]
[sessionlog.txt](https://github.com/arkypita/LaserGRBL/files/7287166/sessionlog.txt)
Hi,sir
attaching log for you...
thanks!

**Comment by augustenlee** [comment]
I think maybe the reason is the CNC mode of the firmware..

**Comment by augustenlee** [comment]
Okay, I will try to parsing the "F:400.0,102.0"
thank you very much..

**Comment by augustenlee** [comment]
Hi, Sir,
thank you very much,
I thought that I found the problem already.
![截圖 2021-10-06 上午1 09 17](https://user-images.githubusercontent.com/15098382/136070432-347d0c43-ccbb-4af9-bb13-fb4ed5f5d39f.png)
.

and the answer from my machine is :
<Idle|MPos:0.0000,0.0000,0.0000|WPos:0.0000,0.0000,0.0000|F:400.0,102.0>

**Comment by arkypita** [comment]
The "`<Idle,MPos:0.000,0.000,0.000,WPos:0.000,0.000,0.000>`" is the code to handle status report from grbl version <= 0.9

If your firmware reply with <Idle|MPos:0.0000,0.0000,0.0000|WPos:0.0000,0.0000,0.0000|F:400.0,102.0> it should be a v1.1

What grbl version is printed out when you connect the engraver?

**Comment by arkypita** [comment]
I believe that your LaserGRBL never receive a grbl "welcome message" so he doesn't know what version is it. This is how welcome message look like

![image](https://user-images.githubusercontent.com/8782035/136080262-85a6e184-7452-48ea-b125-a1cbed8a5908.png)

**Comment by augustenlee** [comment]
> I believe that your LaserGRBL never receive a grbl "welcome message" so he doesn't know what version is it. This is how welcome message look like
> 
> ![image](https://user-images.githubusercontent.com/8782035/136080262-85a6e184-7452-48ea-b125-a1cbed8a5908.png)

No, I haven't seen this information since my connection type was set to "Smoothie"
but thanks any way,
I will try to connect this night..

Sincerely,

Augusten

**Comment by arkypita** [comment]
I believe that since there is a telnet channel involved, grbl sends its welcome message but this is lost because the PC is not connected yet. Maybe you can force re-sending a welcome message by click on thunder symbol (I don't know if smoothie reply to this command with welcome message).

If it doesn't work you can do this: set Type = GRBL and Protocol Emulator, connect once to the emulator so LaserGRBL sees the message Grbl v1.1 and configures itself for that version. Once he has seen it he should remember it.

**Comment by augustenlee** [comment]
dear Sir, r
after checking my smoothieware’s grbl version, It was not V0.9..
 i modified the code  to decode the smoothie gcode as V1.1e
then it worked!

thank you very much…
